src/VisualStudio/Next/Remote/RemoteHostClientServiceFactory.RemoteHostClientService.cs

```diff
@@ -0,0 +1,178 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Execution;
+using Microsoft.CodeAnalysis.Extensions;
+using Microsoft.VisualStudio.LanguageServices.Implementation.Remote;
+
+namespace Microsoft.VisualStudio.LanguageServices.Remote
+{
+    internal partial class RemoteHostClientServiceFactory
+    {
+        private class RemoteHostClientService : IRemoteHostClientService
+        {
+            private readonly Workspace _workspace;
+            private readonly IDiagnosticAnalyzerService _analyzerService;
+
+            private readonly object _gate;
+
+            private CancellationTokenSource _shutdown;
```

Comment:
_shutdownCancellationTokenSource.  _shutdown is very hard to understand in the code elsewhere as it sounds like a boolean or something.